pennylane/operation.py

```diff
@@ -906,6 +906,50 @@ def __matmul__(self, other):
 
         raise ValueError("Can only perform tensor products between observables.")
 
+    def obs_data(self):
+        r"""Extracts the data from an Observable/Tensor."""
+        obs = Tensor(self).non_identity_obs
+        tensor = set()
+
+        for ob in obs:
+            parameters = tuple(param.tostring() for param in ob.parameters)
+            tensor.add((ob.name, ob.wires, parameters))
+
+        return tensor
+
+    def compare(self, other):
+        r"""Compares two Observables/Tensors/qml.Hamiltonian objects to determine if they are equivalent.
```

Comment:
```suggestion
        r"""Compares the observable with another :class:`~.Observable` :class:`~.Tensor`,
        or :class:`~.Hamiltonian` object to determine if they are equivalent.
```

I would perhaps expand on what you mean by 'equivalent' here. For example,

```python
>>> H1 = qml.Hamiltonian([1, 1], [PauliX(0), PauliX(0)])
>>> H2 = qml.Hamiltonian([2], PauliX(0))
>>> H1.compare(H2)
False
```

Reply:
Similarly, `qml.PauliZ(0) != qml.PauliZ(1) @ qml.Identity(0)`.